src/pages/edge/adobe-journey-optimizer/code-based/tabs/tutorial.md

```diff
@@ -0,0 +1,192 @@
+final Surface surface1 = new Surface("mainActivity#banner");
+final Surface surface2 = new Surface("secondActivity#promotions");
+ 
+final List<Surface> surfaces = new ArrayList<>();
+surfaces.add(surface1);
+surfaces.add(surface2);
+ 
+Messaging.getPropositionsForSurfaces(surfaces, new AdobeCallbackWithError<Map<Surface, List<Proposition>>>() {
+    @Override
+    public void fail(final AdobeError adobeError) {
+        // handle error
+    }
+
+    @Override
+    public void call(Map<Surface, List<Proposition>> propositionsMap) {
+        // get the content for the given surfaces
+        if (propositionsMap == null || propositionsMap.isEmpty()) {
+            // bail early if no propositions are found
+            return;
+        }
+
+        // read surface1 propositions
+        List<Proposition> propositionsForSurface1 = propositionsMap.get(surface1);
+
+        // read surface2 propositions
+        List<Proposition> propositionsForSurface2 = propositionsMap.get(surface2);
+    }
+});
+```
+
+<Variant platform="ios" function="getpropositionsforsurfaces" repeat="2"/>
+
+#### Swift
+
+```swift
+let surface1 = Surface(path: "myView#button")
+let surface2 = Surface(path: "myViewAttributes")
+ 
+Messaging.getPropositionsForSurfaces([surface1, surface2]) { propositionsDict, error in
+    guard error == nil else {
+        /// handle error
+        return
+    }
+ 
+    guard let propositionsDict = propositionsDict, !propositionsDict.isEmpty else {
+        /// bail early if no propositions are found
+        return
+    }
+     
+    /// get the content for the given surfaces
+    if let propositionsForSurface1 = propositionsDict[surface1] {
+        /// read surface1 propositions
+    }
+ 
+    if let propositionsForSurface2 = propositionsDict[surface2] {
+        /// read surface2 propositions
+    }
+}
+```
+
+<Variant platform="android" function="using-propositions" repeat="2"/>
+
+#### Java
+
+```java
+// get the propositions for surface1
+if (propositionsForSurface1 == null || propositionsForSurface1.isEmpty()) {
+    // bail early if no propositions are found for surface1
+    return;
+}
+
+// iterate through items in proposition
+for (final PropositionItem propositionItem: propositionsForSurface1.get(0).getItems()) {
+    if (propositionItem.getSchema() == SchemaType.HTML_CONTENT) {
+        final String htmlContent = propositionItem.getHtmlContent();
+        webView.loadData(htmlContent, "text/html", "UTF-8");
+    }
+}
+```
+
+<Variant platform="ios" function="using-propositions" repeat="2"/>
+
+#### Swift
+
+```swift
+/// get the propositions for surface1
+if let codePropositions: [Proposition] = propositionsDict?[surface1], !codePropositions.isEmpty {
+    /// iterate through items in proposition
+    ForEach(codePropositions.first?.items as? [PropositionItem] ?? [], id:\.itemId) { propositionItem in
+        if propositionItem.schema == .htmlContent {
+            let webView = WebView(htmlString: propositionItem.htmlContent)
```

Comment:
maybe here just show how to retrieve the `.htmlContent`?

how each customer uses the content will ultimately be up to them, so can show retrieving the content then have a comment, something like 
```
// use retrieved html content
```